can't update the API json on this branch due to local build problem that i'm not seeing on engine side

```
$EXPECTORATE=overwrite cargo nextest run --all-features --nocapture -E "test(test_openapi)"
....

  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: rust-lld: error: unable to find library -lpython3.12
          collect2: error: ld returned 1 exit status         
```

--- a/modeling-cmds/openapi/api.json
+++ b/modeling-cmds/openapi/api.json
@@ -91,7 +91,17 @@
             "type": "number",
             "format": "float"
           },
+          "from_edge_reference": {
+            "nullable": true,
+            "description": "Edge reference to use to measure the dimension from If both `from_entity_id` and `from_edge_reference` are provided, `from_edge_reference` takes precedence.",
+            "allOf": [
+              {
+                "$ref": "#/components/schemas/EdgeSpecifier"
+              }
+            ]
+          },
           "from_entity_id": {
+            "nullable": true,
             "description": "Entity to measure the dimension from",
             "type": "string",
             "format": "uuid"
@@ -123,7 +133,17 @@
             "format": "uint32",
             "minimum": 0
           },
+          "to_edge_reference": {
+            "nullable": true,
+            "description": "Edge reference to use to measure the dimension from If both `to_entity_id` and `to_edge_reference` are provided, `to_edge_reference` takes precedence.",
+            "allOf": [
+              {
+                "$ref": "#/components/schemas/EdgeSpecifier"
+              }
+            ]
+          },
           "to_entity_id": {
+            "nullable": true,
             "description": "Entity to measure the dimension to",
             "type": "string",
             "format": "uuid"
@@ -141,12 +161,10 @@
           "dimension",
           "font_point_size",
           "font_scale",
-          "from_entity_id",
           "from_entity_pos",
           "offset",
           "plane_id",
           "precision",
-          "to_entity_id",
           "to_entity_pos"
         ]
       },
@@ -179,7 +197,17 @@
               }
             ]
           },
+          "edge_reference": {
+            "nullable": true,
+            "description": "Edge reference to use to place the annotation leader from If both `entity_id` and `edge_reference` are provided, `edge_reference` takes precedence.",
+            "allOf": [
+              {
+                "$ref": "#/components/schemas/EdgeSpecifier"
+              }
+            ]
+          },
           "entity_id": {
+            "nullable": true,
             "description": "Entity to place the annotation leader from",
             "type": "string",
             "format": "uuid"
@@ -248,7 +276,6 @@
           }
         },
         "required": [
-          "entity_id",
           "entity_pos",
           "font_point_size",
           "font_scale",
@@ -262,7 +289,17 @@
         "description": "Parameters for defining an MBD Feature Tag Annotation state",
         "type": "object",
         "properties": {
+          "edge_reference": {
+            "nullable": true,
+            "description": "Edge reference to use to place the annotation leader from If both `entity_id` and `edge_reference` are provided, `edge_reference` takes precedence.",
+            "allOf": [
+              {
+                "$ref": "#/components/schemas/EdgeSpecifier"
+              }
+            ]
+          },
           "entity_id": {
+            "nullable": true,
             "description": "Entity to place the annotation leader from",
             "type": "string",
             "format": "uuid"
@@ -327,7 +364,6 @@
           }
         },
         "required": [
-          "entity_id",
           "entity_pos",
           "font_point_size",
           "font_scale",

